app/models/submission.rb

```diff
@@ -134,4 +155,26 @@ def can_be_ineligible_for_evaluation
 
     errors.add(:judging_status, "must remain eligible for evaluation when evaluators are assigned")
   end
+
+  def calculate_evaluation_status
+    assigned_evaluators = evaluator_submission_assignments.where(status: :assigned)
```

Comment:
nit: prefer the builtin enum methods
```suggestion
    assigned_evaluators = evaluator_submission_assignments.assigned
```